cmd/imosum/main.go

```diff
@@ -21,30 +23,96 @@ func processFile(path string) {
 	fmt.Printf("%016x  %s\n", hash, path)
 }
 
-func walkDirFunction(path string, dirEntry fs.DirEntry, err error) error {
-   if err != nil {
-      return err
-   }
-   if ! dirEntry.IsDir() {
-      processFile(path)
-   }
-   return nil
+func processWalkFunc(path string, dirEntry fs.DirEntry, err error) error {
+	if err != nil {
+		return err
+	}
+	if !dirEntry.IsDir() {
+		processFile(path)
+	}
+	return nil
+}
+
+func checkFile(path string) {
+	file, err := os.Open(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	malformedLines := 0
+	failedHashes := 0
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, "  ", 2)
+
+		if len(parts) != 2 {
+			malformedLines += 1
+			continue
+		}
+
+		readableResult := "FAILED"
+		hash, err := imohash.SumFile(parts[1])
+		if err != nil {
+			failedHashes += 1
+		} else {
+			hashStr := fmt.Sprintf("%016x", hash)
+
+			if hashStr == parts[0] {
+				readableResult = "OK"
+			} else {
+				failedHashes += 1
+			}
+		}
+
+		fmt.Printf("%s: %s\n", parts[1], readableResult)
+	}
+
```

Comment:
`scanner.Scan()` can be `false` for EOF or errors. To know which, check the scanner error here (after the loop), like:

```
 if err := scanner.Err(); err != nil {
      log.Fatal(err)
  }
```